doc/modules/neural_networks_supervised.rst

```diff
@@ -0,0 +1,255 @@
+   :target: ../auto_examples/neural_networks/plot_elm_diagram.html
+   :align: center
+   :scale: 100%
+
+     Figure 1 : Single-hidden layer feedforward network.
+
+ELM has three main hyper-parameters,
+
+  *  ``weight_scale``: controls the variance of the random projection weights;
+  *  ``n_hidden``: controls the number of hidden layer nodes; and
+  *  ``C``: controls the regularization strength of the output linear model.
+
+``weight_scale`` and ``C`` are similar in that they regularize weight parameters
+such that a higher value means less regularization hence less underfitting (or less bias)
+but potentially more overfitting (or more variance). However, in terms of their
+effect on the single-hidden layer feedforward network, ``weight_scale`` regularizes 
+the input-to-hidden connection weights, whereas ``C`` regularizes the hidden-to-output 
+connection weights. Some examples demonstrating the effect of these two parameters are listed
+below:
+
+.. figure:: ../auto_examples/neural_networks/images/plot_elm_scale_C_and_weight_scale_001.png
+   :target: ../auto_examples/neural_networks/plot_elm_scale_C_and_weight_scale.html
+   :align: center
+   :scale: 100%
+
+.. topic:: Examples:
+
+   * :ref:`example_neural_networks_plot_elm_scale_C_and_weight_scale.py`
+
+Higher ``n_hidden`` provides ELM with a higher capacity to learn complex 
+functions. It allows ELM to have more feature detectors in the hidden 
+layer that characterize the training dataset. 
+
+This implements :ref:`classification <elm_classification>` for classification 
+and :ref:`regression <elm_regression>` for regression. They support both dense
+(``numpy.ndarray`` and convertible to that by ``numpy.asarray``) and
+sparse (any ``scipy.sparse``) sample vectors as input. They also support
+on-line and batch training.
+
+
+.. _elm_classification:
+
+Classification
+==============
+
+:class:`ELMClassifier` supports binary-class, multi-class, and
+multi-label classification on datasets.
+
+Like other classifiers, :class:`ELMClassifier` trains on a dataset using the
+fit method that accepts two input parameters: an array X of size 
+``(n_samples, n_features)``, which represents the training samples, and an array y 
+of integer values of size ``(n_samples)``, which holds the class labels respective
+to the training samples::
+
+    >>> from sklearn.neural_network import ELMClassifier
+    >>> X = [[0, 0], [1, 1]]
+    >>> y = [0, 1]
+    >>> clf = ELMClassifier(random_state=1)
+    >>> clf.fit(X, y)
+    ELMClassifier(C=1, activation='relu', batch_size=None, class_weight=None,
+           n_hidden=100, random_state=1, verbose=False, warm_start=False,
+           weight_scale=1.0)
+
+After training, the model can predict the class of a new sample::
+
+    >>> clf.predict([[2., 2.]])
+    array([1])
+
+Weighted classification
+-----------------------
+
+For imbalanced datasets, giving more weight to classes having less samples than 
+others could help improve accuracy. 
```

Comment:
I am not sure if "accuracy" is right. Also the example below should probably report some scores, like f1 or auc?